hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataStorage.java

```diff
@@ -335,7 +335,7 @@ public VolumeBuilder prepareVolume(DataNode datanode,
     VolumeBuilder builder =
         new VolumeBuilder(this, sd);
     for (NamespaceInfo nsInfo : nsInfos) {
-      location.makeBlockPoolDir(nsInfo.getBlockPoolID(), null);
+      location.makeBlockPoolDir(nsInfo.getBlockPoolID(), datanode.getConf());
```

Comment:
what about `datanode.getDNConf()` here?

Reply:
@Hexiaoqiao Thank you for your response. The method cannot use getDnConf here because the type used inside the method is Configuration, not DNConf.

Reply:
Got it. Make sense to me.  `#datanode.getConf()` return HdfsConfiguration instance which is init-ed when start DataNode actually.